src/Components/AuthDialog/Views/AuthDialogLogin.tsx

```diff
@@ -0,0 +1,198 @@
+                  <Clickable
+                    textDecoration="underline"
+                    onClick={() => {
+                      dispatch({
+                        type: "MODE",
+                        payload: { mode: "ForgotPassword" },
+                      })
+                    }}
+                  >
+                    Forgot Password?
+                  </Clickable>
+                </Text>
+              </Box>
+
+              {values.mode === "OnDemand" && (
+                <Message variant="info">
+                  Your safety and security are important to us. Please check
+                  your email for a one-time authentication code to complete your
+                  login.
+                </Message>
+              )}
+
+              {(values.mode === "TwoFactor" || values.mode === "OnDemand") && (
+                <Input
+                  name="authenticationCode"
+                  title="Authentication Code"
+                  placeholder="Enter an authentication code"
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  autoFocus
+                  error={
+                    touched.authenticationCode && errors.authenticationCode
+                  }
+                />
+              )}
+
+              {status?.error && (
+                <Message variant="error">{status.error}</Message>
+              )}
+
+              <Button
+                type="submit"
+                width="100%"
+                loading={values.mode === "Loading"}
+                disabled={!isValid || !dirty}
+              >
+                Log in
+              </Button>
+
+              <Text variant="xs" textAlign="center" color="black60" my={-1}>
+                or
+              </Text>
+
+              <AuthDialogSocial />
+
+              <Text variant="xs" textAlign="center" color="black60">
+                Don’t have an account?{" "}
+                <Clickable
+                  textDecoration="underline"
+                  onClick={() => {
+                    dispatch({ type: "MODE", payload: { mode: "SignUp" } })
+                  }}
+                >
+                  Sign up.
+                </Clickable>
+              </Text>
+            </Join>
+          </Form>
+        )
+      }}
+    </Formik>
+  )
+}
+
+const validationSchema = Yup.object().shape({
+  email: Yup.string()
+    .email("Please enter a valid email.")
+    .required("Email required."),
+  // We allow any input for password since requirements have changed over the years.
+  password: Yup.string().required("Password required"),
+  // If auth code then require it
+  authenticationCode: Yup.string()
+    .when("mode", {
+      is: "OnDemand",
+      then: Yup.string().required("Authentication code required."),
+    })
+    .when("mode", {
+      is: "TwoFactor",
+      then: Yup.string().required("Authentication code required."),
+    }),
```

Comment:
We previously weren't validating this stuff.

Reply:
I'm also wondering if I should validate the format. @joeyAghion can we expect this to always be a numeric sequence?

Reply:
Nope! One-time codes are 6-digit numerics, but backup codes are longer and alphanumeric. You could do some minimal validation of minimum length though.